File: eu.openanalytics.phaedra.tutorials.scripting/2 DilutionFactor/dilutionFactor.js
var plateId = parseInt(args[0]);
var dilutionFactor = parseFloat(args[1]);

var plate = API.get("PlateService").getPlateById(plateId);
var wells = plate.getWells().toArray();
for (var i in wells) {
	var newConc = wells[i].getCompoundConcentration() / dilutionFactor;
	wells[i].setCompoundConcentration(newConc);
}

API.get("PlateService").updatePlate(plate);
API.get("CalculationService").calculate(plate);

File: eu.openanalytics.phaedra.tutorials.scripting/2 DilutionFactor/dilutionFactor.py
plateId = int(str(args.get(0)))
dilutionFactor = float(str(args.get(1)))

plate = API.get("PlateService").getPlateById(plateId)
wells = plate.getWells().toArray()

for well in wells:
	newConc = well.getCompoundConcentration() / dilutionFactor
	well.setCompoundConcentration(newConc)

API.get("PlateService").updatePlate(plate)
API.get("CalculationService").calculate(plate)